### ISSUE TITLE
Standardise .podspec file and its location

### ISSUE THREAD

**Issue by IljaDaderko** [react-native-community/meta#15]
Those of us including third party packages via CocoaPods, require `.podspec` file to be present in the repo.

Atm this is not consistent as this file is missing or located in different places i.e. `ios/` folder or root of the repo.

**Comment by kelset** [comment]
cc @orta @alloy do you have preferences? I think it was something discussed during the new autolink implementation you worked on during the past summit

**Comment by orta** [comment]
It needs to be in the root of the project, not in an iOS folder or it won't work with remote repos (e.g. only via node_modules, which isn't great). We should only be recommending it be in the root.

**Comment by matt-oakes** [comment]
I believe most of them already are, but it's useful for them to be driven from the `package.json` like this:

https://github.com/react-native-community/react-native-netinfo/blob/5fc5379218cdfca7e9eb1dac4a4b517740775161/react-native-netinfo.podspec

**Comment by orta** [comment]
Agree, I think even the git repo url can come from the `package.json` too 👍

**Comment by alloy** [comment]
Perhaps there’s a possibility to create a `pod lib create` template that has a `package.json` and creates the spec in a way that most values are derived from the `package.json`? https://guides.cocoapods.org/making/using-pod-lib-create.html

**Comment by alloy** [comment]
I wouldn’t even use the word “recommend”. A pod repo that will work with all of CP’s options for how to install _must_ have the podspec at the root.

**Comment by grabbou** [comment]
Yeah, I guess the issue should be more about "recommended format of Podfile" that uses "package.json" to pull most of the informations.

I plan to work on the CLI command to scaffold a new library, we could make that follow React Native Community patterns too, so that the entire ecosystem eventually becomes similar to some degree.

**Comment by pvinis** [comment]
Also, a way to soft enforce/indicate the podfile and readme and other "guidelines" would be to list them in the table of repos with ✔️ or ❌.
